packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -144,7 +144,14 @@ export class WorkspaceCommandContribution implements CommandContribution {
                 }
             })
         }));
+        let rootUri: URI | undefined;
+        this.workspaceService.root.then(root => {
+            if (root) {
+                rootUri = new URI(root.uri);
+            }
+        });
         registry.registerCommand(WorkspaceCommands.FILE_DELETE, this.newMultiUriAwareCommandHandler({
+            isVisible: uris => !(rootUri && uris.some( uri => uri.toString() === rootUri!.toString())),
```

Comment:
There is a formatting whitespace before the `uri` parameter. Plus, you could save the root URI as a string after resolving it, so that there is no need to call `toString()` multiple times.

But please hold on with the changes, I need to verify it especially because of your modifications in the `uri-command-handler`.

Reply:
I think the logic is broken. Or at least the type signature doesn't match.

Reply:
I am OK changing it but then we need to get rid of `isValid`. Did you check other use-sites whether it breaks anything else?

Reply:
+1 I removed the isValid stuff.

Reply:
> Plus, you could save the root URI as a string after resolving it, so that there is no need to call toString() multiple times.

This is not a hot path, so I prefer it to be more readable (symmetrical).

Reply:
:+1

Reply:
> Did you check other use-sites whether it breaks anything else?
Yes